### ISSUE TITLE
4.x does not report notices despite E_ALL

### ISSUE THREAD

**Issue by kkmuffme** [stayallive/wp-sentry#79]
I just updated to 4.0.1 but it seems that for PHP only fatal errors are reported by sentry.

I have `define( 'WP_SENTRY_ERROR_TYPES', E_ALL );` set.

e.g. in 3.11.1 for following code:

```
echo $helloworld;

call_this();
exit;
```

I would get 2 errors in sentry:
>Undefined variable: helloworld
>Call to undefined function call_this()

However in 4.0.1, sentry.io only shows:
>Call to undefined function call_this()

Not sure if this is an issue by wp-sentry-integration or the underlying SDK though.

**Comment by stayallive** [comment]
This has possibly to do with the `error_reporting` level you have set.

Sentry uses that to determine which errors to report.

So you should set that to `error_reporting(E_ALL)` too or set the `capture_silenced_errors` option in the [Sentry client option](https://github.com/stayallive/wp-sentry#wp_sentry_options-array).

**Comment by kkmuffme** [comment]
Nope, it doesn't, bc it works fine in 3.11.1. The issue only occurs in 4.0.1, not in 3.11.1 before.

Additionally I have (since ages) set in php.ini/www.conf
`error_reporting = E_ALL`
`php_admin_value[error_reporting] = E_ALL`

and additionally also `define( 'WP_SENTRY_ERROR_TYPES', E_ALL );`

Did you try to reproduce it?

**Comment by stayallive** [comment]
I did, and I could.

However I also do think I know why I was not seeing what you are seeing.

I am running PHP 7.4 and 8 in my testing environment where an undefined variable is a `E_WARNING` on PHP 8 instead of a `E_NOTICE` on 7.4. 

And WordPress set's a default `error_reporting` level with `WP_DEBUG` off (so your `php.ini` settings are overwritten by WordPress which causes confusion.

I am looking into this if this is a SDK bug or we should workaround this in the plugin to make sure we show all the errors the user requested in `error_types` which we currently don't unless `capture_silenced_errors` is true or `error_reporting()` includes the error level.

I'll update here once I know more.

**Comment by kkmuffme** [comment]
Yes, I'm on PHP 7.4

I have had `WP_DEBUG` set to false during all tests.
Since my WP version (& literally everything else in my test environment) is the same when I tested 4.01/3.11.1, I don't think it is anything with WP (or its error handling) itself.

**Comment by stayallive** [comment]
I think you might have unearthed a bug in the Sentry SDK and I also think I have a fix for it, hopefully this makes it into a release soon and we can fix this behaviour: https://github.com/getsentry/sentry-php/pull/1183.

**Comment by stayallive** [comment]
Hi @kkmuffme, it took some time but could you try out the `4.2.0` release that was just released to see if it fixes the issue?

**Comment by kkmuffme** [comment]
Thanks works again correctly with 4.2.0 @stayallive

**Comment by stayallive** [comment]
Perfect, thanks for the feedback 👍

**Comment by lubo-makky** [comment]
@stayallive I am following this issue as I have run into same problem. After fix I upgraded to 4.2.0 however `E_USER_NOTICE` is still not being logged, `E_USER_WARNING and `E_USER_ERROR` are getting thru. With version 3.11.1 reporting of notices worked well, as reported in https://github.com/stayallive/wp-sentry/issues/79#issuecomment-774694958
 
I use PHP 7.4 and test via trigger_error:

```
 trigger_error('Test WP Sentry plugin catching Notices', E_USER_NOTICE);
```

In my wp-config, I've got `define('WP_SENTRY_ERROR_TYPES', E_ALL);`

Is there something I missed?  
Thank you

**Comment by stayallive** [comment]
@lubo-makky I have tested it with `define( 'WP_SENTRY_ERROR_TYPES', E_ALL );` and the following code:

```php
trigger_error( 'Test WP Sentry deprecated', E_USER_DEPRECATED );
trigger_error( 'Test WP Sentry notice', E_USER_NOTICE );
trigger_error( 'Test WP Sentry warning', E_USER_WARNING );
trigger_error( 'Test WP Sentry error', E_USER_ERROR );
```

I got 4 messages in Sentry so I think it is working as intended, I tested this both on PHP 7.4 and 8.0.

Are you sure the `trigger_error` occurs after the Sentry plugin was loaded?

Maybe open up a new issue with some more information about your code and environment so we can investigate?
